dynamic-checkbox-item/load_slack_channels_in_form_question.js

```diff
@@ -0,0 +1,80 @@
+var IGNORED_CHANNELS = [
+  'chat',
+  'slack-questions'
+]
+var FORM_ID = ''
+var SLACK_API_CHANNELS_URL = ''
+
+function onOpen () {
+  var form = FormApp.openById(FORM_ID)
+  var requestResponse = UrlFetchApp.fetch(SLACK_API_CHANNELS_URL)
+  var parsedJsonResponse = JSON.parse(requestResponse.getContentText())
+  var isSlackResponseOk = parsedJsonResponse.ok
+
+  if (!isSlackResponseOk) return
+
+  var eligibleChannels = prepareEligibleChannels(parsedJsonResponse)
+
+  var checkBoxItem = sanitizedCheckBoxItem(form)
+
+  eligibleChannels = eligibleChannels.map(function (channel) {
+    return checkBoxItem.createChoice(channel.name)
+  })
+
+  checkBoxItem
+    .setTitle('Select the channels you will probably join:')
+    .setChoices(eligibleChannels)
+}
+
+function prepareEligibleChannels (parsedJsonResponse) {
+  return parsedJsonResponse.channels.reduce(function (filteredChannels, channel) {
+    if (isChannelEligible(channel)) {
+      filteredChannels.push({
+        name: channel.name,
+        topic: channel.topic.value
+      })
+    }
+    return filteredChannels
+  }, [])
+}
+
+function isChannelEligible (channel) {
+  return !channel.is_general
+    && !channel.is_archived
+    && !channel.is_private
+    && channel.is_channel
+    && !isIgnoredChannel(channel)
+}
+
+function isIgnoredChannel (channel) {
+  return IGNORED_CHANNELS.indexOf(channel.name) > -1
+}
+
+function sanitizedCheckBoxItem (form) {
+  var checkBoxItem = findFirstItem(form)
```

Comment:
@alexstyl Yeap that is easily doable, yet this file needs to be modified in case someone wants to modify an existing item or add a new one each time. Since this function is supposed to be triggered in `onOpen`, in order to always have the latest data, adding a new `CheckBoxItem` would cause a mess on the form. 

In our use case for the invitation form, we will only manipulate the same item.